app/script/util/DebugUtil.js

```diff
@@ -58,6 +58,44 @@ z.util.DebugUtil = class DebugUtil {
       .then(() => this.logger.log(`Corrupted Session ID '${sessionId}'`));
   }
 
+  haveISentThisMessageToMyOtherClients(
+    messageId,
+    conversationId = wire.app.repository.conversation.active_conversation().id
+  ) {
+    let amountOfMessagesSent = 0;
+    let dateTime = undefined;
+
+    const clientId = wire.app.repository.client.currentClient().id;
+    const userId = wire.app.repository.user.self().id;
+
+    const isOTRMessage = notification => notification.type === z.event.Backend.CONVERSATION.OTR_MESSAGE_ADD;
+    const isInCurrentConversation = notification => notification.conversation === conversationId;
+    const wasSentByOurCurrentClient = notification =>
+      notification.from === userId && (notification.data && notification.data.sender === clientId);
+    const hasExpectedTimestamp = notification => notification.time === dateTime.toISOString();
+
+    return wire.app.repository.conversation
+      .get_conversation_by_id(conversationId)
+      .then(conversation => {
+        return wire.app.repository.conversation.get_message_in_conversation_by_id(conversation, messageId);
+      })
+      .then(message => {
+        dateTime = new Date(message.timestamp());
+        return wire.app.repository.event.notificationService.getNotifications(undefined, undefined, 10000);
```

Comment:
One idea to fix the temporal coupling from above, would be to return the message in a `then` of the `getNotification`.

Something like 

```
getNotification(...).then(({ notification }) => ({ notification, message }))
```

Reply:
Then the next function gets the message inject and can check stuff in it